spec/faraday/rack_builder_spec.rb

```diff
@@ -151,6 +151,34 @@ class Banana < Handler
     end
   end
 
+  context 'when adapter is added with named options' do
+    after { Faraday.default_adapter_options = nil }
+    let(:conn) { Faraday::Connection.new {} }
+
+    let(:cat_adapter) do
+      Class.new(Faraday::Adapter) do
+        attr_accessor :name
+
+        def initialize(app, name:)
+          super(app)
+          @name = name
+        end
+      end
+    end
+
+    let(:cat) { subject.adapter.build }
+
+    it 'adds a handler to construct adapter with named options' do
+      Faraday.default_adapter = :cat_adapter
+      Faraday.default_adapter_options = { name: 'Chloe' }
+      Faraday::Adapter.register_middleware(cat_adapter: cat_adapter)
```

Comment:
It would be better not to register the dummy adapter, as that will remain registered for other tests as well. This should work as well:
```suggestion
      Faraday.default_adapter = cat_adapter
      Faraday.default_adapter_options = { name: 'Chloe' }
```

Reply:
I fixed it in e82f273 . I've confirmed that all tests have passed 😃 .
Thank you!!